cpp/bench/ann/src/faiss/faiss_gpu_benchmark.cu

```diff
@@ -138,5 +153,18 @@ REGISTER_ALGO_INSTANCE(std::uint8_t);
 
 #ifdef ANN_BENCH_BUILD_MAIN
 #include "../common/benchmark.hpp"
-int main(int argc, char** argv) { return cuvs::bench::run_main(argc, argv); }
+int main(int argc, char** argv)
+{
+  rmm::mr::cuda_memory_resource cuda_mr;
+  // Construct a resource that uses a coalescing best-fit pool allocator
+  // and is initially sized to half of free device memory.
+  rmm::mr::pool_memory_resource<rmm::mr::cuda_memory_resource> pool_mr{
+    &cuda_mr, rmm::percent_of_free_device_memory(50)};
+  // Updates the current device resource pointer to `pool_mr`
+  auto old_mr = rmm::mr::set_current_device_resource(&pool_mr);
+  auto ret = cuvs::bench::run_main(argc, argv);
+  // Restores the current device resource pointer to its previous value
+  rmm::mr::set_current_device_resource(old_mr);
+  return ret;
+}
```

Comment:
We've run in many issues in ann-bench in the past doing the program-wide setting of memory resource. This also changes the behavior depending on whether `ANN_BENCH_BUILD_MAIN` flag is enabled. Therefore, it's best if these lines stay unmodified and same for all algorithms.
Please move this resource setting to the place where the index is constructed/destructed  in the algo header file.

Reply:
Solved this. I'm now using the configured_raft_resources object similar to how the other cuvs algos are doing.

Reply:
Marking as resolved.